tool/tctl/common/plugins_command.go

```diff
@@ -0,0 +1,106 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package common
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/gravitational/trace"
+
+	pluginsv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/plugins/v1"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/service/servicecfg"
+)
+
+// PluginsCommand allows for management of plugins.
+type PluginsCommand struct {
+	config     *servicecfg.Config
+	cleanupCmd *kingpin.CmdClause
+	pluginType string
+	dryRun     bool
+}
+
+// Initialize creates the plugins command and subcommands
+func (p *PluginsCommand) Initialize(app *kingpin.Application, config *servicecfg.Config) {
+	p.config = config
+
+	pluginsCommand := app.Command("plugins", "Manage Teleport plugins.").Hidden()
+	p.cleanupCmd = pluginsCommand.Command("cleanup", "Cleans up the given plugin type.")
+	p.cleanupCmd.Arg("type", "The type of plugin to cleanup.").StringVar(&p.pluginType)
+	p.cleanupCmd.Flag("dry-run", "Dry run the cleanup command.").BoolVar(&p.dryRun)
+
+}
+
+// Cleanup cleans up the given plugin.
+func (p *PluginsCommand) Cleanup(ctx context.Context, clusterAPI *auth.Client) error {
+	needsCleanup, err := clusterAPI.PluginsClient().NeedsCleanup(ctx, &pluginsv1.NeedsCleanupRequest{
+		Type: p.pluginType,
+	})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	if needsCleanup.PluginActive {
+		fmt.Printf("Plugin of type %q is currently active, can't cleanup!\n", p.pluginType)
+		return nil
+	}
+
+	if !needsCleanup.NeedsCleanup {
+		fmt.Printf("Plugin of type %q doesn't need a cleanup!\n", p.pluginType)
+		return nil
+	}
+
+	if p.dryRun {
+		fmt.Println("Would be deleting the following resources:")
+	} else {
+		fmt.Println("Deleting the following resources:")
+	}
+
+	for _, resource := range needsCleanup.ResourcesToCleanup {
+		fmt.Printf("- %s\n", resource)
+	}
+
+	if p.dryRun {
+		fmt.Println("Since dry run is indicated, I won't delete anything.")
```

Comment:
nit: not sure how I feel about the `I` here. I couldn't find anywhere else we refer `I` in the CLI.

Reply:
I'll reword this.